ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -1505,24 +1505,40 @@ static const char* GetRelativePath(const char* fullPath, const char* root)
         return "";
     }
     
+    string altPath(fullPath);
+    string altRoot(root);
+    const char* removePart = "/System/Volumes/Data";
+    const char* realFullPath;
+    if (altPath.find(removePart) == 0 && altRoot.find(removePart) != 0)
+    {
+        // If fullPath contains "/System/Volumes/Data", but the root does not remote it for comparison
+        // "/System/Volumes/Data/Users/account" is effectively "/Users/account" and should be compared as such
+        realFullPath = fullPath + strlen(removePart);
```

Comment:
Since `removePart` is known at compile time we can avoid having to do a runtime check of its length:

`const char* removePart = "/System/Volumes/Data";` -> `const char removePart[] = "/System/Volumes/Data";`
`strlen(removePart)` -> `(extent<decltype(removePart)>::value - 1)`


_Example_
```
    const char removePart[] = "/System/Volumes/Data";
    printf("String: %s, strlen: %zu, extent(-1 for null): %lu\n",
        removePart,
        strlen(removePart),
        std::extent<decltype(removePart)>::value - 1);
```
_Output_
```
String: /System/Volumes/Data, strlen: 20, extent(-1 for null): 20
```

Reply:
As an alternative to `extent<decltype(removePart)>::value` you could also use `Array_Size` in `ArrayUtilities.hpp`:

```
template<typename T, size_t size>
    constexpr size_t Array_Size(T (&array)[size])
{
    return size;
}
```